### PR TITLE
CASSANDRA-15447 - Pass correct seed node through to InstanceConfig so it can set it properly when starting clusters on non-0 subnets.

### PR DESCRIPTION
**Pull request by JeetKunDoug**

### CODE REVIEW

File: test/distributed/org/apache/cassandra/distributed/impl/InstanceConfig.java [hunk 1]
```diff
@@ -239,20 +240,21 @@ public String getString(String name)
...
+    public static InstanceConfig generate(int nodeNum, String ipAddress, NetworkTopology networkTopology, File root, String token, String seedIp)
     {
         return new InstanceConfig(nodeNum,
                                   networkTopology,
                                   ipAddress,
                                   ipAddress,
                                   ipAddress,
                                   ipAddress,
-                                  String.format("%s/node%d/saved_caches", root, nodeNum),
+                seedIp, String.format("%s/node%d/saved_caches", root, nodeNum),
```
Comment (yifan-c) [inline test/distributed/org/apache/cassandra/distributed/impl/InstanceConfig.java]: nit: one line per parameter

Comment (JeetKunDoug) [reply test/distributed/org/apache/cassandra/distributed/impl/InstanceConfig.java]: Also weird indentation - will fix up.

File: test/distributed/org/apache/cassandra/distributed/impl/InstanceConfig.java [hunk 2]
```diff
@@ -239,20 +240,22 @@ public String getString(String name)
...
                                   ipAddress,
                                   ipAddress,
+                                  seedIp,
                                   String.format("%s/node%d/saved_caches", root, nodeNum),
                                   new String[] { String.format("%s/node%d/data", root, nodeNum) },
                                   String.format("%s/node%d/commitlog", root, nodeNum),
                                   String.format("%s/node%d/hints", root, nodeNum),
                                   String.format("%s/node%d/cdc", root, nodeNum),
-                                  token);
+                                  token
```
Comment (dcapwell) [inline test/distributed/org/apache/cassandra/distributed/impl/InstanceConfig.java]: nit: moving it back would shrink the diff (up to you)

Comment (JeetKunDoug) [reply test/distributed/org/apache/cassandra/distributed/impl/InstanceConfig.java]: fixed

**Pull request closed without merging**